exp/api/v1beta2/rosaroleconfig_types.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ROSARoleConfigSpec defines the desired state of ROSARoleConfig
+
+type ROSARoleConfigSpec struct {
+	AccountRoleConfig  AccountRoleConfig  `json:"accountRoleConfig"`
+	OperatorRoleConfig OperatorRoleConfig `json:"operatorRoleConfig"`
+	OIDCConfig         OIDCConfig         `json:"oidcConfig"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=rosaroleconfig,scope=cluster,categories=cluster-api,shortName=rosarole
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// ROSARoleConfig is the Schema for the rosaroleconfigs API
+type ROSARoleConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ROSARoleConfigSpec   `json:"spec,omitempty"`
+	Status ROSARoleConfigStatus `json:"status,omitempty"`
+}
+
+type AccountRoleConfig struct {
+	// TODO: verify max len
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +optional
+	Path string `json:"path,omitempty"`
+	// +kubebuilder:validation:Required
+	Version string `json:"version"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type OperatorRoleConfig struct {
+	// TODO: verify max len
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +kubebuilder:validation:Required
+	OIDCConfigID string `json:"oidcConfigId,omitempty"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type SharedVPCConfig struct {
+	RouteRoleARN       string `json:"routeRoleARN,omitempty"`
+	VPCEndpointRoleARN string `json:"vpcEndpointRoleArn,omitempty"`
+}
+
+type OIDCConfig struct {
+	CreateManagedOIDC     string                 `json:"createManagedOIDC"`
+	ExternalAuthProviders []ExternalAuthProvider `json:"externalAuthProviders,omitempty"`
+}
+
+type ExternalAuthProvider struct {
```

Comment:
lets use the API defined with the control plane https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controlplane/rosa/api/v1beta2/external_auth_types.go

Reply:
We cant import it because it would create circular dependency. SO I have created copy in `/exp`  directory